### ISSUE TITLE
AmiGO autocomplete: exlude obsolete terms

### ISSUE THREAD

**Issue by pgaudet** [geneontology/amigo#454]
Hello, 

We'd like to exclude obsolete terms from the autocomplete feature of AmiGO. 
Obsolete terms should be searchable (at least by ID) but it would be less confusing if they were not shown in the autocomplete. 

Thanks, Pascale

**Comment by kltm** [comment]
There is currently no data in the general index for obsoletion. While the UI parts are easy, we'll have to add an `obsolete` field to the index and populate that appropriately for ontology classes. It shouldn't be difficult, but will have to be coordinated with an embedded release of owltools as well as a schema update.

**Comment by cmungall** [comment]
> We'd like to exclude obsolete terms from the autocomplete feature of AmiGO.
> Obsolete terms should be searchable (at least by ID) but it would be less confusing if they were not shown in the autocomplete.

I agree they should remain searchable. I would say there is no reason to restrict it to ID. Often a user may have in mind a term that they perceive to be in scope for GO but which we think is not, and have obsoleted. E.g histolysis. We should allow them to search for the concept and to arrive at an explanation of why we have not included as a true term.

We should have very clear behavior around obsoletes in search

 - obsolete results should be downweighted (ideally this would happen naturally, if we downweight less used terms)
 - there should be a clear visual indication of obsoletion status. Currently we do this in the name by prefixing with "obsolete", which is good, as it guarantees we have some indication across all tools and portals, but UIs should do more, e.g strikethrough
 
Regarding behavior in autocomplete. I think it's confusing to have different logic between search and AC. The user searching for an obsolete concept will expect AC results. However, given the smaller set of terms returned in AC, and the lack of context, the recommendations above are even more important, to avoid obsoletes flooding results. An obsolete should never be prioritized over a non-obsolete (perhaps except in the case of an exact match)

> There is currently no data in the general index for obsoletion. While the UI parts are easy, we'll have to add an obsolete field to the index and populate that appropriately for ontology classes. It shouldn't be difficult, but will have to be coordinated with an embedded release of owltools as well as a schema update.

I think we really need to do this
